### ISSUE TITLE
🚀 Feature: increment id for Documentid?

### ISSUE THREAD

**Issue by anandsubbu007** [appwrite/appwrite#3001]
### 🔖 Feature description

is it possible to set auto increment id (integer) or increment is with some prefix sufix?

### 🎤 Pitch

i want to store invoice id in order with custom prefix with increment id

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by Meldiron** [comment]
Yes, this is possible 😇 What you need to do is to create Appwrite Function, and in settings, hook it to the document creation event. Next, you write a function that picks the Id of the last document, increment it, and sets a new ID on the document.

The function will now trigger automatically, and the attribute with incrementing number will be automatically filled within less than a second by the function.

**Comment by BenMemescape** [comment]
It would be great to have this built in (for example, specify 'autoinc()' instead of 'unique()' when creating the document). It's something that the underlying MySQL database provides for free, and it does it atomically which is much harder to do with a custom function - for example, the mechanism described by @Meldiron seems vulnerable to a race condition.

**Comment by eldadfux** [comment]
@Meldiron this could work well with the server vars idea we had in mind.

**Comment by felix-d1strict** [comment]
Yes, that would be great if implemented. 🤩

**Comment by cholasimmons** [comment]
'autoinc()' would be lovely!  
How do I upvote this feature

**Comment by stnguyen90** [comment]
> How do I upvote this feature

@cholasimmons, you can add a 👍 on the original [comment](https://github.com/appwrite/appwrite/issues/3001#issue-1178378246) by clicking the emoji icon at the top right of the comment.

**Comment by S-M-Ammar** [comment]
You can use "unique()" as string for auto generated id

Example:

const promise = databases.createDocument('someDatabaseID'', 'SomeCollectionID', "unique()", {});

// Here document id is auto generated as we pass "unique()" string in parameters. 

Hope it helps. Thanks.

**Comment by BenMemescape** [comment]
> You can use "unique()" as string for auto generated id
> 
> Example:
> 
> const promise = databases.createDocument('someDatabaseID'', 'SomeCollectionID', "unique()", {});
> 
> // Here document id is auto generated as we pass "unique()" string in parameters.
> 
> Hope it helps. Thanks.

Thanks for trying, but no that doesn't help. I already mentioned `unique()` above, so you should know we're aware of it. What's being requested here is an automatically incrementing **numeric** ID. So the first one requested for a given collection is 1, the next one is 2, and so on. They should be atomic so that it's impossible for two records within the same collection to get the same auto-incremented ID. The MySQL back end provides this service, but Appwrite doesn't.

`unique()` produces an ID which is a long string of characters - it's a completely different thing.

**Comment by YouSour** [comment]
any update ? need this feature is useful for everyone. 🤔

**Comment by stnguyen90** [comment]
Please also provide your use case and **why** this feature is necessary for people requesting this feature. That would give us much more context for the problem.

**Comment by felix-d1strict** [comment]
Hello @stnguyen90 ,
my use case would be a custom CMS where it would be useful if the URLs were built according to the following scheme:
your.tld/1-appwrite-is-very-good
your.tls/2-auto-increment-is-missing
and so on. Because it is possible that two articles have the same title.

Currently I have to query the number of all documents to determine the number and increment it by 1 when using the create form in my CMS. 

And alternatively using a unique() - document ID is also out of the question, because the string would simply be much too long.

My question would be more why it was never really considered. I mean MySQL and so on already have this function.

**Comment by abnegate** [comment]
@felix-d1strict Would being able to set an integer attribute to auto-increment when creating it solve your issue?

**Comment by felix-d1strict** [comment]
> @felix-d1strict Would being able to set an integer attribute to auto-increment when creating it solve your issue?

Yes, exactly. 👍🏻

**Comment by sergeycherepanov** [comment]
> Please also provide your use case and **why** this feature is necessary for people requesting this feature. That would give us much more context for the problem.

Hi, as an example in my case, I building a self-hosted saas solution where I need to increment the VM id as numerical value (it depends on the hapervisor api). Yes I can use a function for it, but the function can cause collitions in case when it used in parallel executions. In case when sevaral fucntiona will be started in one time, how I can be sure tha it will not select the same value for incrementaiton.

**Comment by stnguyen90** [comment]
This is now available! See https://appwrite.io/blog/post/announcing-auto-increment-support
